src/components/link/link.component.js

```diff
@@ -2,131 +2,126 @@ import React from "react";
-import LinkStyle from "./link.style";
+import { StyledLink, StyledContent } from "./link.style";
 import tagComponent from "../../utils/helpers/tags";
 
-class InternalLink extends React.Component {
-  static safeProps = ["onClick"];
-
-  onKeyDown = (ev) => {
-    if (this.props.onKeyDown) {
-      this.props.onKeyDown(ev);
-    }
-
-    // return early if there is no onClick or there is a href prop
-    // or the event is not an enter key
-    if (this.props.href || (!Event.isEnterKey(ev) && !Event.isSpaceKey(ev))) {
-      return;
-    }
-
-    if (this.props.onClick) {
-      this.props.onClick(ev);
-    }
-  };
-
-  renderLinkIcon = (currentAlignment = "left") => {
-    const hasProperAlignment =
-      this.props.icon && this.props.iconAlign === currentAlignment;
-
-    return hasProperAlignment ? this.icon : null;
-  };
-
-  get icon() {
-    return (
-      <Icon
-        type={this.props.icon}
-        bgTheme="none"
-        iconColor="business-color"
-        disabled={this.props.disabled}
-        ariaLabel={this.props.ariaLabel}
-        tooltipMessage={this.props.tooltipMessage}
-        tooltipPosition={this.props.tooltipPosition}
-      />
-    );
-  }
-
-  get tabIndex() {
-    return this.props.tabbable && !this.props.disabled ? "0" : "-1";
-  }
-
-  get componentProps() {
-    return {
-      onKeyDown: this.onKeyDown,
-      onMouseDown: this.props.onMouseDown,
-      disabled: this.props.disabled,
-      onClick: this.handleClick,
-      tabIndex: this.tabIndex,
-      target: this.props.target,
-      ref: this.props.innerRef,
-      href: this.props.href,
-      rel: this.props.rel,
+const Link = React.forwardRef(
+  (
+    {
+      children,
+      className,
+      onKeyDown,
+      href,
+      onClick,
+      onMouseDown,
+      icon,
+      iconAlign,
+      isSkipLink,
+      disabled,
+      ariaLabel,
+      rel,
+      tooltipMessage,
+      tooltipPosition,
+      tabbable,
+      target,
+      ...props
+    },
+    ref
+  ) => {
+    const tabIndex = tabbable && !disabled ? "0" : "-1";
+    const handleOnKeyDown = (ev) => {
+      if (onKeyDown) {
+        onKeyDown(ev);
+      }
+
+      // return early if there is no onClick or there is a href prop
+      // or the event is not an enter key
+      if (href || (!Event.isEnterKey(ev) && !Event.isSpaceKey(ev))) {
+        return;
+      }
+
+      if (onClick) {
+        onClick(ev);
+      }
+    };
+    const renderLinkIcon = (currentAlignment = "left") => {
+      const hasProperAlignment = icon && iconAlign === currentAlignment;
+
+      return hasProperAlignment ? (
+        <Icon
+          type={icon}
+          bgTheme="none"
+          iconColor="business-color"
+          disabled={disabled}
+          ariaLabel={ariaLabel}
+          tooltipMessage={tooltipMessage}
+          tooltipPosition={tooltipPosition}
+        />
+      ) : null;
+    };
+    const handleClick = (ev) => {
+      /* istanbul ignore if */
```

Comment:
I think we need to be testing this rather than skipping coverage here

Reply:
I have no idea how I can test it. I tried to simulate click and mocked event, but there is no way to simulate click if button is disabled. I've searched for force click but cannot find any way to accomplish this. Do you have any suggestions?

Reply:
So in the browser if it's disabled, does it not trigger this handleClick at all? If so we can just remove the check here

Reply:
probably we can remove it, I'll confirm that, thanks